src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -374,9 +374,10 @@ internal override bool HasPointerType
                 if (_lazyType != null)
                 {
                     Debug.Assert(_lazyType.Value.DefaultType.IsPointerType() ==
-                        IsPointerFieldSyntactically());
+                        IsPointerFieldSyntactically() ||
+                        _lazyType.Value.DefaultType.Kind == SymbolKind.FunctionPointer);
 
-                    return _lazyType.Value.DefaultType.IsPointerType();
+                    return _lazyType.Value.DefaultType.IsPointerType() || _lazyType.Value.DefaultType.Kind == SymbolKind.FunctionPointer;
```

Comment:
>IsPointerType [](start = 55, length = 13)

I assume you also checked other places where we check for regular pointer type, by using this helper or type/symbol kind.

Reply:
Yes. These were the only places I feel comfortable changing at this point. I did mean to add a prototype comment to `IsPointerType` to follow up on some more places, but I apparently missed staging it.

---
In reply to: [376482345](https://github.com/dotnet/roslyn/pull/41455#discussion_r376482345) [](ancestors = 376482345)